### PR TITLE
ref(pkg/kube): extract wait logic from install/update

### PR DESCRIPTION
**Pull request by adamreese**
This change adds a new method for waiting for kubernetes resources.

Also snuck in: use Duration flag types

### CODE REVIEW

File: pkg/action/uninstall.go [hunk 1]
```diff
@@ -220,15 +219,15 @@ func (u *Uninstall) deleteRelease(rel *release.Release) (kept string, errs []err
...
+		kept = summarizeKeptManifests(filesToKeep, u.cfg.KubeClient)
 	}
 
 	for _, file := range filesToDelete {
 		b := bytes.NewBufferString(strings.TrimSpace(file.Content))
 		if b.Len() == 0 {
 			continue
 		}
-		if err := u.cfg.KubeClient.Delete(rel.Namespace, b); err != nil {
+		if err := u.cfg.KubeClient.Delete(b); err != nil {
```
Comment (bacongobbler) [inline pkg/action/uninstall.go]: is there a way we can refactor `.Wait` to accept a condition? That way we could wait for these resources to reach the deleted state. I've heard people ask about that.

Doesn't have to be done in this PR, just thought if you wondered if that's possible

**Review by bacongobbler — COMMENTED** [review]
looks great! I think this is missing `--wait` functionality on `helm upgrade` and `helm rollback`, but once that's addressed we should be good to merge. I really love the refactor work done in `pkg/kube` here. :+1:

File: pkg/action/rollback.go [hunk 2]
```diff
@@ -149,7 +149,8 @@ func (r *Rollback) performRollback(currentRelease, targetRelease *release.Releas
 
 	cr := bytes.NewBufferString(currentRelease.Manifest)
 	tr := bytes.NewBufferString(targetRelease.Manifest)
-	if err := r.cfg.KubeClient.Update(targetRelease.Namespace, cr, tr, r.Force, r.Recreate, r.Timeout, r.Wait); err != nil {
+	// TODO add wait
```
[the remark this answers is not in the document]
Comment (adamreese) [reply pkg/action/rollback.go]: I was going to add the wait logic back in a followup PR.  This just pulls out the wait param from the interface.

File: pkg/action/uninstall.go (hunk 1 above)
Comment (adamreese) [reply pkg/action/uninstall.go]: That is my plan!

**Review by bacongobbler — APPROVED** [review]
All right! The refactor here looks great. Let's get this in.

**Pull request merged**